content/blog/announcing-enum-support/index.md

```diff
@@ -0,0 +1,159 @@
+```
+
+{{< /choosable >}}
+{{< choosable language python >}}
+
+```python
+import pulumi
+from pulumi_aws import s3
+
+my_bucket = s3.Bucket("myBucket", acl="private")
+```
+
+{{< /choosable >}}
+{{< choosable language csharp >}}
+
+```csharp
+using Pulumi;
+using S3 = Pulumi.Aws.S3;
+
+class MyStack : Stack
+{
+    public MyStack()
+    {
+        var myBucket = new S3.Bucket("myBucket", new S3.BucketArgs
+        {
+            Acl = "private",
+        });
+    }
+}
+```
+
+{{< /choosable >}}
+{{< choosable language go >}}
+
+```go
+package main
+
+import (
+    "github.com/pulumi/pulumi-aws/sdk/v3/go/aws/s3"
+    "github.com/pulumi/pulumi/sdk/v2/go/pulumi"
+)
+
+func main() {
+    pulumi.Run(func(ctx *pulumi.Context) error {
+        myBucket, err := s3.NewBucket(ctx, "myBucket", &s3.BucketArgs{
+            Acl: pulumi.String("private"),
+        })
+        if err != nil {
+            return err
+        }
+        return nil
+    })
+}
+```
+
+{{< /choosable >}}
+
+In the above code, the S3 bucket resource has an `acl` property, where we pass in the string `"private"`. However, as the [resource docs](https://www.pulumi.com/docs/reference/pkg/aws/s3/bucket/#acl_nodejs) indicate, the `acl` property can only be set to one of a few different values, namely `"private"`, `"public-read"`, `"public-read-write"`, `"aws-exec-read"`, `"authenticated-read"` and `"log-delivery-write"`. The `acl` property is the perfect candidate for an enum type and is now emitted as one, so you can use the following code instead.
+
+{{< choosable language typescript >}}
+
+```typescript
+import * as pulumi from "@pulumi/pulumi";
+import { s3 } from "@pulumi/aws";
+
+const myBucket = new s3.Bucket("myBucket", { acl: s3.CannedAcl.Private });
+```
+
+{{< /choosable >}}
+{{< choosable language python >}}
+
+```python
+import pulumi
+from pulumi_aws import s3
+
+my_bucket = s3.Bucket("myBucket", acl=s3.CannedAcl.PRIVATE)
+```
+
+{{< /choosable >}}
+{{< choosable language csharp >}}
+
+```csharp
+using Pulumi;
+using S3 = Pulumi.Aws.S3;
+
+class MyStack : Stack
+{
+    public MyStack()
+    {
+        var myBucket = new S3.Bucket("myBucket", new S3.BucketArgs
+        {
+            Acl = S3.CannedAcl.Private,
+        });
+    }
+}
+```
+
+{{< /choosable >}}
+{{< choosable language go >}}
+
+```go
+package main
+
+import (
+    "github.com/pulumi/pulumi-aws/sdk/v3/go/aws/s3"
+    "github.com/pulumi/pulumi/sdk/v2/go/pulumi"
+)
+
+func main() {
+    pulumi.Run(func(ctx *pulumi.Context) error {
+        myBucket, err := s3.NewBucket(ctx, "myBucket", &s3.BucketArgs{
+            Acl: s3.CannedAclPrivate,
+        })
+        if err != nil {
+            return err
+        }
+        return nil
+    })
+}
+```
+
+{{< /choosable >}}
+
+The enum properties provide discoverable and normalized constants that can be used in place of raw strings, allowing you to move faster with the added boosts of IDE tooling like type hints and autocomplete.
```

Comment:
A few thoughts for additions:
1. A short animated GIF (or even just screenshot) of a completion list showing the valid values, with help text - and discussion of the discoverability benefits
2. A screenshot of an error squiggle under a typo'd string value being passed - and discussion of how this helps catches errors *much* faster, instead of waiting until the code is deployed
3. A note on the strict vs. loose enums, and how we've modelled both of these - applying them to different cases to balance flexibility and improved error checking / discoverability.
4. To the point in the next paragraph - perhaps a quick note on where enums are available now (and in what forms - perhaps a list of a few enums for each of AWS and Azure NextGen?), as well as a hint that we'll be adding more enums across the Pulumi SDKs in the coming months.